provider/provider.go

```diff
@@ -785,42 +786,63 @@ func (p *Provider) DispatchWithCtx(ctx context.Context, appPublicKey, chain stri
 }
 
 // Relay does request to be relayed to a target blockchain
-func (p *Provider) Relay(rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, error) {
+func (p *Provider) Relay(rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, *RelayOutputErr) {
 	return p.RelayWithCtx(context.Background(), rpcURL, input, options)
 }
 
 // RelayWithCtx does request to be relayed to a target blockchain
-func (p *Provider) RelayWithCtx(ctx context.Context, rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, error) {
+func (p *Provider) RelayWithCtx(ctx context.Context, rpcURL string, input *RelayInput, options *RelayRequestOptions) (*RelayOutput, *RelayOutputErr) {
 	rawOutput, reqErr := p.doPostRequest(ctx, rpcURL, input, ClientRelayRoute, http.Header{})
 
 	defer closeOrLog(rawOutput)
 
 	if reqErr != nil && !errors.Is(reqErr, errOnRelayRequest) {
-		return nil, reqErr
+		return nil, &RelayOutputErr{Error: reqErr, StatusCode: rawOutput.Status}
 	}
 
-	bodyBytes, err := ioutil.ReadAll(rawOutput.Body)
+	bodyBytes, err := io.ReadAll(rawOutput.Body)
 	if err != nil {
-		return nil, err
+		return nil, &RelayOutputErr{Error: err, StatusCode: rawOutput.Status}
 	}
 
 	if errors.Is(reqErr, errOnRelayRequest) {
-		return nil, parseRelayErrorOutput(bodyBytes, input.Proof.ServicerPubKey)
+		return nil, &RelayOutputErr{Error: parseRelayErrorOutput(bodyBytes, input.Proof.ServicerPubKey), StatusCode: rawOutput.Status}
 	}
 
 	return parseRelaySuccesfulOutput(bodyBytes)
 }
 
-func parseRelaySuccesfulOutput(bodyBytes []byte) (*RelayOutput, error) {
+// TODO: Remove this function after the node respond back to us with a statusCode along side with the response and the signature
+func extractStatusFromResponse(response string) string {
+	patterns := []*regexp.Regexp{
```

Comment:
NIT: I still think this should be global and initialized once in an `init` but not going to block.

CC @adshmh for another POV

Reply:
Yes, I agree. These should only be constructed once: all of these regexps are knowin in advance, and  compiling them as part of processing every request is unnecessary (no strong opinions here on where the one-time initialization should take place)

Reply:
Done